### PR TITLE
--[BUGFIX] Background renderer should not create extra contexts

### PR DESCRIPTION
**Pull request by jturner65**
## Motivation and Context
This PR fixes an issue where an additional GL context was being created in the Background Renderer, which was causing problems, including seg faults with [examples/tutorials/async_rendering.py
](https://github.com/facebookresearch/habitat-sim/blob/main/examples/tutorials/async_rendering.py)

## How Has This Been Tested
Existing C++ and python tests pass

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/gfx/BackgroundRenderer.cpp [hunk 1]
```diff
@@ -181,23 +180,17 @@ int BackgroundRenderer::threadRender() {
...
-      Cr::Containers::pointer<Mn::Platform::GLContext>(Mn::NoCreate);
-  if (!threadContext_->tryCreate())
-    Mn::Fatal{} << "BackgroundRenderer: Failed to create OpenGL context";
+  context_->makeCurrent();
 
-  Mn::GL::Context::makeCurrent(threadContext_.get());
   threadOwnsContext_ = true;
 
-  Renderer::setupMagnumFeatures();
+  // Renderer::setupMagnumFeatures(); // TODO probably also for nothing, drop
```
Comment (mosra) [inline src/esp/gfx/BackgroundRenderer.cpp:193]: if it worked without, remove it altogether

**Comment by 0mdc** [comment]
Is the code from [examples/tutorials/async_rendering.py](https://github.com/facebookresearch/habitat-sim/blob/main/examples/tutorials/async_rendering.py) tested somewhere?

**Review by 0mdc — APPROVED** [review]
LGTM!

**Comment by jturner65** [comment]
> Is the code from [examples/tutorials/async_rendering.py](https://github.com/facebookresearch/habitat-sim/blob/main/examples/tutorials/async_rendering.py) tested somewhere?

Yes, it is part of the python tests. test_example.py

**Pull request merged**